SIL.Archiving/IMDI/Schema/IMDI_3_0.cs

```diff
@@ -1945,10 +1945,24 @@ public void AddFile(ArchivingFile file)
 		}
 
 		/// <remarks/>
-		public void AddFileAccess(string fullFileName, ArchivingPackage package)
+		public void AddFileAccess(string fullFileName, ArchivingPackage package, LanguageString conditions, LanguageString restrictions)
 		{
 			var sessionFile = GetFile(fullFileName);
 			sessionFile.Access = new AccessType(package);
+			if (conditions == null) conditions = new LanguageString{Value = null};
```

Comment:
Would this work?
conditions = conditions ?? new LanguageString { Value = null };